test/unit_tests/braket/default_simulator/openqasm/test_interpreter.py

```diff
@@ -2207,3 +2206,80 @@ def test_measure_invalid_qubit():
 def test_measure_qubit_out_of_range(qasm, expected):
     with pytest.raises(IndexError, match=expected):
         Interpreter().build_circuit(qasm)
+
+
+@pytest.mark.parametrize(
+    "qasm, expected",
+    [
+        (
+            """
+                bit[2] b;
+                qubit["10"] r1;
+                b = measure r1;
+            """,
+            [0, 1],
+        ),
+        (
+            """
+                bit[3] b;
+                qubit["11"] r1;
+                b = measure r1;
+            """,
+            [0, 1, 2],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit[!"1"] r1;
+                b = measure r1;
+            """,
+            [],
+        ),
+        (
+            """
+                qubit["1" ^ "0"] r1;
+            """,
+            [],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit["1" != "0"] r1;
+                b = measure r1;
+            """,
+            [0],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit["1" == "0"] r1;
+                b = measure r1;
+            """,
+            [],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit[1] r1;
+                h r1["0" << "1"];
+                b = measure r1;
+            """,
+            [0],
+        ),
+        (
+            """
+                bit[2] b;
+                qubit[1] r1;
+                h r1["0" >> "1"];
+                b = measure r1;
+            """,
+            [0],
+        ),
+    ],
+)
+def test_circuit_from_string_literal(qasm, expected):
+    try:
+        circ = Interpreter().build_circuit(source=qasm)
+        assert expected == circ.measured_qubits
+    except AttributeError as exc:
+        pytest.fail("Interpreter raised an AttributeError")
```

Comment:
[nit] I'd suggest you remove this `try` / `except`. If the `build_circuit()` call raises an exception, it will cause the test to fail - you don't need to manually call `pytest.fail`.